tests/foreman/virtwho/utils.py

```diff
@@ -180,43 +236,47 @@ def _get_hypervisor_mapping(logs):
         )
 
 
-def deploy_validation(debug=True):
+def deploy_validation():
     """Checkout the deploy result
-    :param bool debug: if VIRTWHO_DEBUG=0, this option should be False,
-        because there is no json data logged in rhsm.log.
     :raises: VirtWhoError: If failed to start virt-who servcie.
+    :ruturn: hypervisor_name and guest_name
     """
     status = get_virtwho_status()
     _, logs = runcmd('cat /var/log/rhsm/rhsm.log')
     error = len(re.findall(r'\[.*ERROR.*\]', logs))
     if status != 'running' or error != 0:
         raise VirtWhoError("Failed to start virt-who service")
-    if debug:
-        hypervisor_name, guest_name = _get_hypervisor_mapping(logs)
-        # Delete the hypervisor entry and always make sure it's new.
-        hosts = Host.list({'search': hypervisor_name})
-        for index, host in enumerate(hosts):
-            Host.delete({'id': host['id']})
-        runcmd("systemctl restart virt-who")
-        return hypervisor_name, guest_name
+    hypervisor_name, guest_name = _get_hypervisor_mapping(logs)
+    # Delete the hypervisor entry and always make sure it's new.
+    hosts = Host.list({'search': hypervisor_name})
+    for index, host in enumerate(hosts):
+        Host.delete({'id': host['id']})
```

Comment:
why are you enumerating here?

Reply:
Yeah, it should be dropped because we don't need to use the `index` in the current method.